openmdao/recorders/test/test_dump.py

```diff
@@ -422,6 +423,38 @@ def test_driver_records_metadata(self):
 
         self.assertMetadataRecorded((expected_params, expected_unknowns, expected_resids))
 
+    def test_driver_records_unknown_types_metadata(self):
+        
+        prob = Problem()
+        root = prob.root = Group()
+
+        # Need an optimization problem to test to make sure 
+        #   the is_desvar, is_con, is_obj metadata is being 
+        #   recorded for the Unknowns
+        root.add('p1', IndepVarComp('x', 50.0), promotes=['*'])
+        root.add('p2', IndepVarComp('y', 50.0), promotes=['*'])
+        root.add('comp', Paraboloid(), promotes=['*'])
```

Comment:
Agreed that we need a story to get rid of all of them, but we also need to not add any new ones, because unfortunately fixing them isn't going to be just a simple global search and replace.  Somebody is going to have to look at each one and see what vars are in the component and add those manually, so please fix these and help out the poor soul who eventually has to do that story.  :)